File: src/main/java/org/example/backend/account/repository/AccountRepository.java
package org.example.backend.account.repository;

import java.util.Optional;
import org.example.backend.account.entity.Account;
import org.springframework.data.jpa.repository.JpaRepository;

public interface AccountRepository extends JpaRepository<Account, Long> {

    Optional<Account> findByKakaoOauthId(Long kakaoOauthId);
}

File: src/main/kotlin/sharev/account/adapter/outbound/jpa/repository/AccountRepository.kt
package sharev.account.adapter.outbound.jpa.repository

import org.springframework.data.jpa.repository.JpaRepository
import sharev.account.adapter.outbound.jpa.entity.AccountJpaEntity

interface AccountRepository : JpaRepository<AccountJpaEntity, Long> {
    fun findByHandle(handle: String): AccountJpaEntity?

    fun existsByHandleAndIdNot(handle: String, id: Long): Boolean
}
